### PR TITLE
Make a completely new website

### PR DESCRIPTION
**Pull request by DopustimVladimir**
This website is absolutely new. It was made to look modern and feel better.

Pros:

- Dark as coffee
- Responsive layout
- Brand new logo
- Markdown-based pages
- Improved Cake build
- Simple and clean code

Cons:

- Need to specify the new location of the "index.html" and other files in the Github settings
- I have not edited any files in the ".github" directory

Any comments welcome

### CODE REVIEW

**Comment by UziTech** [comment]
Looks great :100:.

The only thing that would be needed is for coffeelint version and rules to be updated automatically on build. Currently whenever coffeelint is updated this site is automatically updated with any new/changed rules or rule information and version.

**Comment by DopustimVladimir** [comment]
@UziTech Can we use GitHub Actions for this purpose? It seems that we can just run `npm update` before `npm run build`. And of course we can update dependencies from "package.json" to `"@coffeelint/cli": "latest"`. Or we can use the latest versions "coffeescript.js" and "coffeelint.js" directly from the repo in HTML. Or we can get it from Cake, like currently. I'm not sure what's the right way...

**Comment by UziTech** [comment]
You can see how we currently do it in the [cakefile](https://github.com/coffeelint/coffeelint.github.io/blob/532899f78e770d1dde3e9851ef20a86310e3d129/Cakefile#L15).

Basically, every coffeelint update uses GitHub actions to run the update command in this repo.

**Comment by DopustimVladimir** [comment]
@UziTech ok, done

**Comment by UziTech** [comment]
Do the rules automatically get updated in rules.md if one were to get added?

**Comment by DopustimVladimir** [comment]
@UziTech thanks for noticing! Now this table with rules will be built automatically

**Comment by sibiraj-s** [comment]
Conflicts. can we not commit the build directory? we can use github actions to deploy to `gh-pages` branch. 

Suggested action https://github.com/peaceiris/actions-gh-pages. 

This is for another PR.

**Comment by UziTech** [comment]
Ya the merge conflicts need to be fixed so this can be merged. I just want to make sure this site is still completely automated. Anytime coffeelint is updated this site should update coffeelint and coffeescript for the repl and update the rules list automatically.

**Comment by sibiraj-s** [comment]
Should be possible. We can use github api's to trigger action in another repo and setup workflow dispatch here to receive the event. Authentication needs to be configured.


@UziTech . I will fireup a PR for automating that. can you provide me access to pages settings.

**Comment by sibiraj-s** [comment]
> Should be possible. We can use github api's to trigger action in another repo and setup workflow dispatch here to receive the event. Authentication needs to be configured.
> 

This seems to be already configured.

**Comment by UziTech** [comment]
> This seems to be already configured.

Yes this is currently how the site works. I want to make sure it will still work with the new site in this PR.

**Comment by UziTech** [comment]
> can you provide me access to pages settings.

@sibiraj-s I gave you admin rights so you should have access to settings now.

**Comment by sibiraj-s** [comment]
Awesome. Thanks 👍

**Comment by DopustimVladimir** [comment]
@sibiraj-s I can resolve this conflict in my repo, but I cannot merge pr (only those with write access to this repository can merge pull requests). During this time, the bot can make commits, so get ready to check

**Comment by sibiraj-s** [comment]
I will have to check this tomorrow

**Comment by DopustimVladimir** [comment]
@sibiraj-s Thanks! I removed everything and the LICENSE.md file was added out of habit. Some text has been used (partly), but that text is not a software. What the project owner will say? We can type (c) CoffeeLint Team, but you didn't take me to the team 😅. Also we can just close this PR for now

**Comment by sibiraj-s** [comment]
Sorry I didn't mean to say anything wrong. Not used to get LICENSE's over PR's. We don't need LICNSE file at all IMO.

**Comment by sibiraj-s** [comment]
I am done with the review.

**Comment by UziTech** [comment]
I'm ok with the license.

**Comment by DopustimVladimir** [comment]
I like this solution to resolve this softly. We can keep all of stuff at once. Any suggestions are welcome...

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,2 @@
-node_modules
-*.log
-.node-version
+build/
```
Comment (UziTech) [inline .gitignore:1]: I'm pretty sure the built files need to be in the repo to be served by github pages

**Comment by sibiraj-s** [comment]
> I'm pretty sure the built files need to be in the repo to be served by github pages

we can configure actions to deploy github pages. not necessary to be included in the source

**Comment by sibiraj-s** [comment]
I can go ahead and merge this. I can fix the suggestions myself. I'll anyway to make some changes to test github pages deploy actions.

File: .gitignore (hunk 1 above)
Comment (DopustimVladimir) [reply .gitignore:1]: Ok. We can do it another way, without this breaking changes

**Comment by DopustimVladimir** [comment]
Ok, guys. it seems that I made a mistake with this big breaking PR. Sorry about that. I will go another way to keep the website works without any conflicts and will send a new better PR later

**Comment by sibiraj-s** [comment]
@DopustimVladimir This PR is well and good. whats happened? I don't see anything wrong here.

**Pull request closed without merging**